doc/src/guides/solving/solving-guidance.md

```diff
@@ -0,0 +1,141 @@
+# Solving Guidance
+
+Here are guidelines that apply to many types of solving.
+
+## Equations with no analytical solution
+
+The vast majority of arbitrary nonlinear equations are not analytically 
+solvable. The classes of equations that are solvable are basically:
+1. Linear equations
+2. Polynomials, except where limited by the Abel-Ruffini theorem
+3. Equations that can be solved by inverting some transcendental functions
+4. Problems that can be transformed into the cases above 
+(e.g., by turning trigonometric functions into polynomials)
+5. A few other special cases that can be solved with something like the
+Lambert W function
+
+SymPy may reflect that your equation has no solutions that can be expressed 
+algebraically (symbolically) by returning an error such as 
+`NotImplementedError`:
+
+```py
+>>> from sympy import solve, cos
+>>> from sympy.abc import x
+>>> solve(cos(x) - x, x)
+Traceback (most recent call last):
+  ...
+NotImplementedError: multiple generators [x, cos(x)]
+No algorithms are implemented to solve equation -x + cos(x)
+```
+
+so you may have to {func}`solve your equation numerically
+<sympy.solvers.solvers.nsolve>` instead, for example
+
+```py
+>>> from sympy import nsolve, cos
+>>> from sympy.abc import x
+>>> nsolve(cos(x) - x, x, 2)
+0.739085133215161
+```
+
+If you receive non-closed-form solutions, you can evaluate them numerically 
+using {func}`evalf n <sympy.core.evalf.EvalfMixin.n>`:
+
+```py
+>>> from sympy import solve
+>>> from sympy.abc import x
+>>> solutions = solve(x**5 - x - 1, x, dict=True)
+>>> print(solutions)
+[{x: CRootOf(x**5 - x - 1, 0)}, {x: CRootOf(x**5 - x - 1, 1)}, {x: CRootOf(x**5 - x - 1, 2)}, {x: CRootOf(x**5 - x - 1, 3)}, {x: CRootOf(x**5 - x - 1, 4)}]
+>>> [solution[x].n(3) for solution in solutions]
+[1.17, -0.765 - 0.352*I, -0.765 + 0.352*I, 0.181 - 1.08*I, 0.181 + 1.08*I]
```

Comment:
Rather than "non-closed form" it would be better to say "RootOf" or something more specific.

Reply:
It can also be useful to use `evalf` even if the result is closed form:
```python
solve(x**4 + 10*x**2 + x + 1, x)
```
The `solve` function will generally try to return exact solutions if its inputs are exact. Sometimes exact numeric solutions can have very complicated representations though but as long as they are numeric (do not involve symbols) then evalf should be able to reduce them to a simple albeit approximate form.

Of course if you did know that you had a polynomial and therefore that your solution was going to come out in terms of RootOf then solve is not the right function anyway: `nroots`, `real_roots`, `Poly.all_roots`, `roots`. We have many options for solving a univariate polynomial with different strengths and weaknesses.

RootOf specifically as well as the Abel-Ruffini theorem etc is something that deserves special mention in the section on polynomials but could be linked to from other places because the same properties apply general nonlinear equation solvers like solve and nonlinsolve whenever polynomials arise in the equations.